types/components/Collapse.d.ts

```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import { TransitionCallbacks } from './helpers';
 
-interface CollapseProps
+export interface CollapseProps
```

Comment:
This interface is missing the `id` prop seen in [the examples](https://react-bootstrap.github.io/components/navbar/#navbars-overview).

Reply:
It's also missing className. 

#3446 

![image](https://user-images.githubusercontent.com/7609893/52744665-53010680-2fab-11e9-9638-9a92cfe11580.png)

Reply:
I noticed that too but it's not listed in the react-bootstrap docs, and also seems to work in some cases without the id prop

Reply:
@rvaidya The id prop is non-mandatory. Add these please 😄 

```js
id?: string;
className?: string;
```

@thejohnfreeman I believe adding these 2 lines should do it 👍

Reply:
Shouldn't this extend the base html props instead of hand adding these?

Reply:
Yeah I was gonna look into that since it wasn't done originally

Reply:
I've been wondering - is there a base props type for HTML elements? What is it called? Where does it come from? `@types/react`?

Related: is there a base props type for React elements (with `key` and `children`)?

Reply:
There are yeah, in the react types on DT

Reply:
the base proptype I see is React.HTMLProps<T>, which has been applied to some classes but not others.  Some of these decisions seem to be intentional - such as the conflict between "size" in Button:

size?: 'sm' | 'lg'

vs:

size?: number in React.HTMLProps

Based on this, I can just add id?: string to Collapse without extending anything, although the documentation for react-bootstrap should be updated to include id if the id attribute is important

Reply:
@rvaidya Adding the baseProps will also add the declarations for other "not-required" variables and functions. We should avoid that since the documentation doesn't say so, nor the supported methods have been added or extended by the class.

Best to add `id` and `className` in directly. 

Thanks @jquense and @thejohnfreeman for the support 👍

Reply:
Oops sorry `id` doesn't belong here. The screenshot above is for `NavbarCollapse` which specifically renders a `div` the abstract Collapse doesn't render anything but it's child.

Reply:
@jquense But the type declarations of it exist in `Collapse.d.ts` no?

Reply:
it should exist in `NavbarCollapse.d.ts` because its specific to `NavbarCollapse`, but it should be more general to `div` props not just id

Reply:
its fine to not fix this in this PR but `id` shouldn't be added here

Reply:
@jquense I tired replacing the NavbarCollapse definition to

![image](https://user-images.githubusercontent.com/7609893/52816071-65904400-306e-11e9-80c5-961155b52fed.png)
 
but it results in 

![image](https://user-images.githubusercontent.com/7609893/52816112-89538a00-306e-11e9-90b9-3a91fb402017.png)

Reply:
I would just defer this for now

Reply:
@jquense Giving this one last try. Let me know if you're okay with this.... It fixes the issue. Otherwise someone else can take this in the future.

```js
interface NavbarCollapseProps extends React.ClassAttributes<NavbarCollapse> {
  id?: string;
  className?: string;
}

declare class NavbarCollapse extends BsPrefixComponent<typeof Collapse, NavbarCollapseProps> {}
```

Thanks for the continued support on this 😄

Reply:
- `NavbarCollapse` renders a `Collapse` with a `div` child, and passes all its props besides `bsPrefix` through to the `Collapse`.
- The `Collapse` renders a `Transition` with a child render function that renders the child of `Collapse` (the `div`). `Collapse` takes out the `className` prop, passing it through to its child. It passes its remaining unknown props through to the `Transition`, which is imported from `react-transition-group`.
- `Transition` subtracts its props and passes the rest back to its render child. `Collapse` defines a render child that passes these props through to its child (the `div` in this case).

So it seems that `Collapse` should expect to get a mix of the props it knows it uses (already [listed as properties in `CollapseProps`](https://github.com/react-bootstrap/react-bootstrap/blob/b6651f3507ab51c63c84e13ac58d0ae0f084bf0e/types/components/Collapse.d.ts)), the props expected by `Transition` (named `TransitionCallbacks`, which `CollapseProps` already extends), and the props of its child (the `div` in this case, but could be anything). This last type is conceptually `PropsOf<children>`, but I don't know how to express that in TypeScript.

Reply:
ya I think what we actually need to do long term is change the API here so each component doesn't just pass through everything to it's children

Reply:
@prateekkathal , I'm with @jquense and think you should leave the type fixing out of this PR, especially since it's so complicated. Just getting proper exports is a great step forward.

Reply:
Sounds good. 👍 

@rvaidya Please revert the change here 😄